src/SayMore/Utilities/FileSyncHelper.cs

```diff
@@ -0,0 +1,322 @@
+		{
+			switch (client)
+			{
+				case SyncClient.Dropbox:
+					if (!_stoppedDropbox)
+					{
+						KillProcess(kDropboxProcess);
+						_stoppedDropbox = true;
+					}
+					break;
+
+				case SyncClient.GoogleDrive:
+					if (!_stoppedGoogleDrive)
+					{
+						KillProcess(kGoogleDriveProcess);
+						_stoppedGoogleDrive = true;
+					}
+					break;
+
+				case SyncClient.OneDrive:
+					if (!_stoppedOneDrive)
+					{
+						// OneDrive has an off switch
+						var exeFile = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "Microsoft", "OneDrive", "OneDrive.exe");
+						if (File.Exists(exeFile))
+						{
+							LaunchWithCmd(exeFile, "/shutdown");
+							_stoppedOneDrive = true;
+						}
+					}
+					break;
+			}
+		}
+
+		/// <summary>
+		/// Restart the sync clients stopped by Saymore
+		/// </summary>
+		public static void RestartAllStoppedClients()
+		{
+			if (_stoppedDropbox)
+			{
+				var exeFile = GetProgramFilesPath(Path.Combine("Dropbox", "Client", "Dropbox.exe"));
+
+				if (!string.IsNullOrEmpty(exeFile))
+				{
+					LaunchWithCmd(exeFile);
+					_stoppedDropbox = false;
+				}
+			}
+
+			if (_stoppedGoogleDrive)
+			{
+				var exeFile = GetProgramFilesPath(Path.Combine("Google", "Drive", "googledrivesync.exe"));
+
+				if (!string.IsNullOrEmpty(exeFile))
+				{
+					LaunchWithCmd(exeFile);
+					_stoppedGoogleDrive = false;
+				}
+			}
+
+			if (_stoppedOneDrive)
+			{
+				var exeFile = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "Microsoft", "OneDrive", "OneDrive.exe");
+
+				if (File.Exists(exeFile))
+				{
+					// the '/background' switch stops it from opening the OneDrive directory in Explorer
+					LaunchWithCmd(exeFile, "/background");
+					_stoppedOneDrive = false;
+				}
+			}
+		}
+
+		/// <summary>
+		/// Checks if the selected sync client is currently running
+		/// </summary>
+		/// <param name="client"></param>
+		/// <returns>true id the client is running, false if not</returns>
+		public static bool ClientIsRunning(SyncClient client)
+		{
+			string processNameLower;
+
+			switch (client)
+			{
+				case SyncClient.None:
+					return false;
+
+				case SyncClient.Dropbox:
+					processNameLower = kDropboxProcess;
+					break;
+
+				case SyncClient.GoogleDrive:
+					processNameLower = kGoogleDriveProcess;
+					break;
+
+				case SyncClient.OneDrive:
+					processNameLower = kOneDriveProcess;
+					break;
+
+				default:
+					return false;
+			}
+
+			var processes = Process.GetProcesses();
+			return processes.Any(p => p.ProcessName.ToLower() == processNameLower);
+		}
+
+		/// <summary>
+		/// First checks if the file is in a potentially synced directory,
+		/// then if it is, checks if the sync program is active.
+		/// </summary>
+		/// <param name="pathParts"></param>
+		/// <param name="client"></param>
+		/// <returns></returns>
+		private static bool IsThisClient(string[] pathParts, SyncClient client)
```

Comment:
Might want to come up with a name that more accurately suggests that its job is also to see if the client is active.